### PR TITLE
refactor(lint): move lint rule docs to cli/tools/lint/docs/

### PR DESCRIPTION
**Pull request by bartlomieju**
Also adds an internal (hidden) flag to get a list of all rules,
and adds tests to ensure that all lint rules have docs as
well as the schema files are up to date.

### CODE REVIEW

File: cli/schemas/lint-tags.v1.json
Comment (bartlomieju) [inline cli/schemas/lint-tags.v1.json:1]: This is not yet automated and tested, waiting on https://github.com/denoland/deno/pull/27162 to land so appropriate APIs are available.

File: cli/tools/lint/docs/adjacent_overload_signatures.md
Comment (dsherret) [inline cli/tools/lint/docs/adjacent_overload_signatures.md:1]: I don't think we should store these docs here. They're docs for lint rules that live in deno_lint, so they should either live there or we should move all these docs to the deno-docs repo.

File: cli/args/flags.rs [hunk 1]
```diff
@@ -302,6 +302,7 @@ pub struct LintFlags {
   pub json: bool,
   pub compact: bool,
   pub watch: Option<WatchFlags>,
+  pub internal_print_all_rules: bool,
```
Comment (dsherret) [inline cli/args/flags.rs:305]: I don't think we need this? We can add a unit test that does this behaviour in tools/lint?

Comment (bartlomieju) [reply cli/args/flags.rs:305]: Not until https://github.com/denoland/deno/pull/27245 lands, there's no way to get all available rules right now and print them.

Comment (dsherret) [reply cli/args/flags.rs:305]: I think we can use `LintRuleProvider`? https://github.com/denoland/deno/blob/d99b2d6f7db1f88e353021ed55e9492a3997ed8e/cli/tools/lint/rules/mod.rs#L146

**Comment by bartlomieju** [comment]
Closing in favor of https://github.com/denoland/deno/pull/27324 and another PR that will be opened in `deno-docs` repo.

**Pull request closed without merging**